tests/test_odilo.py

```diff
@@ -0,0 +1,342 @@
+        )
+        self.api = self.circulation.api_for_collection[self.collection.id]
+
+        self.edition, self.licensepool = self._edition(
+            data_source_name=DataSource.ODILO,
+            identifier_type=Identifier.ODILO_ID,
+            collection=self.collection,
+            identifier_id=self.RECORD_ID,
+            with_license_pool=True
+        )
+
+    @classmethod
+    def sample_data(cls, filename):
+        return sample_data(filename, 'odilo')
+
+    @classmethod
+    def sample_json(cls, filename):
+        data = cls.sample_data(filename)
+        return data, json.loads(data)
+
+    def error_message(self, error_code, message=None, token=None):
+        """Create a JSON document that simulates the message served by
+        Odilo given a certain error condition.
+        """
+        message = message or self._str
+        token = token or self._str
+        data = dict(errorCode=error_code, message=message, token=token)
+        return json.dumps(data)
+
+
+class TestOdiloCirculationAPI(OdiloAPITest):
+    #################
+    # General tests
+    #################
+
+    # Test 404 Not Found --> patron not found --> 'patronNotFound'
+    def test_01_patron_not_found(self):
+        patron_not_found_data, patron_not_found_json = self.sample_json("error_patron_not_found.json")
+        self.api.queue_response(404, content=patron_not_found_json)
+
+        assert_raises(PatronNotFoundOnRemote, self.api.checkout, '123456789', self.PIN, self.licensepool, 'ACSM')
+        print 'Test patron not found ok!'
+
+    # Test 404 Not Found --> record not found --> 'ERROR_DATA_NOT_FOUND'
+    def test_02_data_not_found(self):
+        data_not_found_data, data_not_found_json = self.sample_json("error_data_not_found.json")
+        self.api.queue_response(404, content=data_not_found_json)
+
+        self.licensepool.identifier.identifier = '12345678'
+        assert_raises(NotFoundOnRemote, self.api.checkout, self.PATRON, self.PIN, self.licensepool, 'ACSM')
+        print 'Test resource not found on remote ok!'
+
+    #################
+    # Checkout tests
+    #################
+
+    # Test 400 Bad Request --> Invalid format for that resource
+    def test_11(self):
+        self.api.queue_response(400, content="")
+        assert_raises(NoAcceptableFormat, self.api.checkout, self.PATRON, self.PIN, self.licensepool, 'FAKE_FORMAT')
+        print 'Test invalid format for resource ok!'
+
+    def test_12_checkout_acsm(self):
+        checkout_data, checkout_json = self.sample_json("checkout_acsm_ok.json")
+        self.api.queue_response(200, content=checkout_json)
+        self.checkout('ACSM')
+
+    def test_13_checkout_ebook_streaming_epub(self):
+        checkout_data, checkout_json = self.sample_json("checkout_ebook_streaming_ok_epub.json")
+        self.api.queue_response(200, content=checkout_json)
+        self.checkout('EBOOK_STREAMING')
```

Comment:
I would call this helper method something like `validate_checkout` so that it's clear that it contains assertions.

Reply:
Ok, I've renamed it to 'perform_and_validate_checkout'.